lib/utils.js

```diff
@@ -0,0 +1,46 @@
+var Q = require('q');
+var es = require('.././config/es');
+
+var log = console.log;
+
+exports = module.exports = {
+  waitForElastic: function (waitCnt) {
+    var waitCntDef = 50;
+    waitCnt = waitCnt || 50;  // 40 requests by default
+
+    var deferred = Q.defer();
+
+    (function pinger(done) {
+      if(waitCnt != waitCntDef) log('requests left', waitCnt);
+
+      if(waitCnt-- == 0) {
+        deferred.reject(new Error('Max requests exceeded, elastic is dead.'));
```

Comment:
My first attempt was the same :D
The problem with this is that elastic ignores timeout (which leads to fast execution of all these requests, even with 500 requests), and in some cases second argument to `then` (I noticed only when `catch` occurs). I didn't have much time to dig into elastic on the subject :(
But maybe I miss something, I'll check this version and let you know whether I'm correct.